packages/twenty-server/src/modules/messaging/message-folder-manager/services/sync-message-folders.service.ts

```diff
@@ -0,0 +1,189 @@
+import { Injectable } from '@nestjs/common';
+
+import { ConnectedAccountProvider } from 'twenty-shared/types';
+import { isDefined } from 'twenty-shared/utils';
+import { v4 } from 'uuid';
+
+import { FeatureFlagService } from 'src/engine/core-modules/feature-flag/services/feature-flag.service';
+import { type WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';
+import { WorkspaceRepository } from 'src/engine/twenty-orm/repository/workspace.repository';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { type MessageChannelWorkspaceEntity } from 'src/modules/messaging/common/standard-objects/message-channel.workspace-entity';
+import { type MessageFolderWorkspaceEntity } from 'src/modules/messaging/common/standard-objects/message-folder.workspace-entity';
+import { GmailGetAllFoldersService } from 'src/modules/messaging/message-folder-manager/drivers/gmail/gmail-get-all-folders.service';
+import { ImapGetAllFoldersService } from 'src/modules/messaging/message-folder-manager/drivers/imap/imap-get-all-folders.service';
+import { MicrosoftGetAllFoldersService } from 'src/modules/messaging/message-folder-manager/drivers/microsoft/microsoft-get-all-folders.service';
+import { MessageFolderName } from 'src/modules/messaging/message-import-manager/drivers/microsoft/types/folders';
+
+type SyncMessageFoldersInput = {
+  workspaceId: string;
+  messageChannelId: string;
+  manager: WorkspaceEntityManager;
+};
+
+type MessageFolder = Pick<
+  MessageFolderWorkspaceEntity,
+  'name' | 'isSynced' | 'isSentFolder' | 'externalId'
+>;
+
+@Injectable()
+export class SyncMessageFoldersService {
+  constructor(
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    private readonly featureFlagService: FeatureFlagService,
+    private readonly gmailGetAllFoldersService: GmailGetAllFoldersService,
+    private readonly microsoftGetAllFoldersService: MicrosoftGetAllFoldersService,
+    private readonly imapGetAllFoldersService: ImapGetAllFoldersService,
+  ) {}
+
+  async syncMessageFolders(input: SyncMessageFoldersInput): Promise<void> {
+    const { workspaceId, messageChannelId, manager } = input;
+
+    const messageChannelRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<MessageChannelWorkspaceEntity>(
+        workspaceId,
+        'messageChannel',
+      );
+
+    const messageChannel = await messageChannelRepository.findOne(
```

Comment:
looks like what we really need is connectedAccount? I feel we could avoid querying data on db if it's not used